extensions/amp-lightbox-gallery/0.1/amp-lightbox-gallery.js

```diff
@@ -223,7 +224,13 @@ export class AmpLightboxGallery extends AMP.BaseElement {
         'i-amphtml-element');
     const clonedNode = element.cloneNode(deepClone);
     clonedNode.removeAttribute('on');
-    clonedNode.removeAttribute('id');
+    const id = clonedNode.getAttribute('id');
```

Comment:
I would infinitely prefer not to require this where possible. But afaik there are ads that require an `id` attribute to work. Can I ask which types of ads require ids? Should I just keep ids for all `<amp-ad>` components? There should be a better heuristic for noting lightbox-gallery children whose ids should be preserved.

Reply:
I know the background to this.

**In short:** we should not do this and no "shippable ad in prod" requires id.

**In long:** `fake-network` is a "demo"-only mode of `amp-ad` for serving A4A's that are not coming from an Ad server. Since we did not want folks to use `fake-network` in production, a requirement was added to have `id=i-amphtml-demo-` on the element which makes the whole page invalid.

**Solution:** Don't use `fake-network` in your Demo page. Use an actual A4A ad from DFP .
e.g.
```
<amp-ad data-slot="/30497360/a4a/a4a_native"
  height="250"
  type="doubleclick"
  width="300">
</amp-ad>
```